llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -19791,15 +19792,63 @@ static SDValue performIntrinsicCombine(SDNode *N,
           EVT::getVectorVT(*DAG.getContext(), PromVT.getVectorElementType(),
                            VT.getVectorElementCount());
 
-      Res = DAG.getNode(ISD::INTRINSIC_WO_CHAIN, DL, WhileVT, ID,
-                        N->getOperand(1), N->getOperand(2));
+      SDValue Res = DAG.getNode(ISD::INTRINSIC_WO_CHAIN, DL, WhileVT, ID,
+                                N->getOperand(1), N->getOperand(2));
       Res = DAG.getNode(ISD::SIGN_EXTEND, DL, PromVT, Res);
       Res = DAG.getNode(ISD::EXTRACT_SUBVECTOR, DL, ExtVT, Res,
                         DAG.getConstant(0, DL, MVT::i64));
       Res = DAG.getNode(ISD::TRUNCATE, DL, VT, Res);
+
+      return Res;
     }
-    return Res;
+
+    if (!Subtarget->hasSVE2p1() && !Subtarget->hasSME2())
+      return SDValue();
+
+    if (!N->hasNUsesOfValue(2, 0))
```

Comment:
Why does the intrinsic call have two uses? The tests only seem to suggest one use. Perhaps worth adding a comment here explaining why.

Reply:
I believe these are the extract that the N has for when we have the mask value as nxv32i8. That is the reason he tests extracts in lanes 19817-19823.

Reply:
Ah ok, in that case I think it's just the commit structure isn't right then because this code was added in the first commit, but there are no tests for it. It looks like we're adding code that in one commit that is only being tested in another. For example, in the first commit there is the test get-active-lane-mask-32x1.ll, which looks like it cannot pass because there is only one use of the intrinsic result in IR.